vendor/github.com/containerd/containerd/content/proxy/content_writer.go

```diff
@@ -125,7 +125,8 @@ func (rw *remoteWriter) Commit(ctx context.Context, size int64, expected digest.
 
 	rw.digest = resp.Digest
 	rw.offset = resp.Offset
-	return nil
+
+	return rw.Close()
```

Comment:
This should be also called in `defer`

Reply:
(On second though, defer close might not be needed)

Reply:
I think commit should be "finalized" even if an error is returned